src/crawlee/_request.py

```diff
@@ -275,7 +278,7 @@ class Request(BaseRequestData):
     @classmethod
     def from_url(
         cls,
-        url: str,
+        url: str | URL,
```

Comment:
To be honest, I'm not a fan of this approach. It feels like a slippery slope—why not accept even more types and simply convert them to strings here? Unless there's a particular reason to accept `yarl.URL`, I would stick to accepting just strings and let users explicitly call it with the conversion like `Request.from_url(str(my_url))`.